### PR TITLE
optimize_CI

### PR DESCRIPTION
**Pull request by SunsetWolf**
## How Has This Been Tested?

- [ ] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [ ] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by ChiahungTai** [comment]
yes.....
The CI is very annoying.... 
Make people away from contribution.
The furstated part

File: .github/workflows/test_qlib_from_source.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
     # References for parameters: https://github.com/PyCQA/pylint/issues/4577#issuecomment-1000245962
     - name: Check Qlib with pylint
       run: |
-        pylint --disable=C0104,C0114,C0115,C0116,C0301,C0302,C0411,C0413,C1802,R0401,R0801,R0902,R0903,R0911,R0912,R0913,R0914,R0915,R1720,W0105,W0123,W0201,W0511,W0613,W1113,W1514,E0401,E1121,C0103,C0209,R0402,R1705,R1710,R1725,R1735,W0102,W0212,W0221,W0223,W0231,W0237,W0612,W0621,W0622,W0703,W1309,E1102,E1136 --const-rgx='[a-z_][a-z0-9_]{2,30}$' qlib --init-hook "import astroid; astroid.context.InferenceContext.max_inferred = 500"
+        pylint --disable=C0104,C0114,C0115,C0116,C0301,C0302,C0411,C0413,C1802,R0401,R0801,R0902,R0903,R0911,R0912,R0913,R0914,R0915,R1720,W0105,W0123,W0201,W0511,W0613,W1113,W1514,E0401,E1121,C0103,C0209,R0402,R1705,R1710,R1725,R1735,W0102,W0212,W0221,W0223,W0231,W0237,W0612,W0621,W0622,W0703,W1309,E1102,E1136 --const-rgx='[a-z_][a-z0-9_]{2,30}$' qlib --init-hook "import astroid; astroid.context.InferenceContext.max_inferred = 500; import sys; sys.setrecursionlimit(2000)"
```
Comment (you-n-g) [inline .github/workflows/test_qlib_from_source.yml:92]: `setrecursionlimit`  is hard to understand,  please add comments about  why it is necessary

File: qlib/tests/config.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def get_data_handler_config(
 
 def get_dataset_config(
     dataset_class=DATASET_ALPHA158_CLASS,
-    train=("2008-01-01", "2014-12-31"),
+    train=("2013-01-01", "2014-12-31"),
```
Comment (you-n-g) [inline qlib/tests/config.py:71]: Make sure this will not affect others  https://github.com/microsoft/qlib/search?q=get_dataset_config

**Comment by you-n-g** [comment]
@SunsetWolf 
It looks great now!  
Thanks!

@ChiahungTai The CI is fixed now...

**Comment by ChiahungTai** [comment]
> @SunsetWolf It looks great now! Thanks!
> 
> @ChiahungTai The CI is fixed now...

Thanks for the info. :)

**Pull request merged**